### PR TITLE
feat: "rejectionhandled" Web event and "rejectionHandled" Node event

### PR DESCRIPTION
**Pull request by bartlomieju**
This commit adds support for "rejectionhandled" Web Event and "rejectionHandled" Node event.

```js
import process from "node:process";

process.on("rejectionHandled", (promise) => {
  console.log("rejectionHandled", reason, promise);
});

window.addEventListener("rejectionhandled", (event) => {
  console.log("rejectionhandled", event.reason, event.promise);
});
```

### CODE REVIEW

File: runtime/js/99_main.js [hunk 1]
```diff
@@ -341,9 +341,13 @@ function runtimeStart(
     tsVersion,
   );
   core.setBuildInfo(target);
+  // TODO(bartlomieju): should be done during snapshot time like
+  // core.setUnhandledPromiseRejectionHandler
+  core.setHandledPromiseRejectionHandler(processRejectionHandled);
```
Comment (bartlomieju) [inline runtime/js/99_main.js:346]: Blocked by https://github.com/denoland/deno_core/pull/430

**Comment by bartlomieju** [comment]
This change will get rid of all these warnings about not implemented "rejectionHandled" event in Next.js demo.

File: cli/tests/testdata/node/rejection_handled_web_process.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
+import chalk from "npm:chalk";
+import process from "node:process";
+
+console.log(chalk.red("Hello world!"));
```
Comment (mmastrac) [inline cli/tests/testdata/node/rejection_handled_web_process.ts:4]: Is this just so we have an npm dep?

**Review by mmastrac — APPROVED** [review]
LGTM

File: cli/tests/testdata/node/rejection_handled_web_process.ts (hunk 2 above)
Comment (bartlomieju) [reply cli/tests/testdata/node/rejection_handled_web_process.ts:4]: Yup

**Review by littledivy — APPROVED** [review]
LGTM

**Pull request merged**